### ISSUE TITLE
Cannot find module 'routing-controllers'.

### ISSUE THREAD

**Issue by lorthirk** [pleerock/routing-controllers#53]
Hey there. I'm trying to use routing-controllers in my project; however, after following the instructions in README.md, I get a `error TS2307: Cannot find module 'routing-controllers'.` wherever I try to use it, e.g. `import { Controller, Param, Body, Get, Post, Put, Delete, Req, Res } from "routing-controllers";``

I'm on node v7 and TypeScript 2.0.10. Not sure what I am doing wrong though... routing-controllers folder is correctly in my node_modules folder.

**Comment by pleerock** [comment]
hmmm thats strange, I guess its something with your environment setup. Try to checkout [sample](https://github.com/pleerock/routing-controllers-express-demo) and run it.

**Comment by eduardoweiland** [comment]
I have found the same problem. I am migrating a project from JS to TypeScript, and I had the `allowJs` config enabled in my tsconfig.json:

```
{
  "compilerOptions": {
    "allowJs": true,
    // other options
  }
}
```

I don't know why this only happens when using `routing-controllers`. I am already using TypeORM with the `allowJs` option enabled without problems. This error only happened after installing this package.

**Comment by pleerock** [comment]
hmmm this issue needs more info, and maybe a git repo with minimal reproduction of the problem which I can checkout and test on

**Comment by eduardoweiland** [comment]
For my case, steps to reproduce:

```
$ git clone https://github.com/pleerock/routing-controllers-express-demo.git
$ cd routing-controllers-express-demo
$ npm install
# edit tsconfig.json as below
$ tsc
```

Error when running tsc:

```
src/app.ts(2,67): error TS2307: Cannot find module 'routing-controllers'.
src/controllers/CategoryController.ts(1,62): error TS2307: Cannot find module 'routing-controllers'.
src/controllers/PostController.ts(1,74): error TS2307: Cannot find module 'routing-controllers'.
```

tsconfig.json diff:
```diff
 {
     "version": "1.8.0",
     "compilerOptions": {
+        "allowJs": true,
         "outDir": "build",
         "target": "es6",
         "module": "commonjs",
         "moduleResolution": "node",
         "emitDecoratorMetadata": true,
         "experimentalDecorators": true,
         "sourceMap": true,
-        "noImplicitAny": true,
-        "declaration": true
+        "noImplicitAny": true
     },
     "exclude": [
         "build",
         "node_modules"
     ]
 }
```

I had to remove `"declaration": true` option too because it cannot be used with `allowJs`.

**Comment by eduardoweiland** [comment]
I found [this page](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) that says to set `types` option in the library's `package.json` (not the app's package.json). I have tested here, adding `"types": "index.d.ts"` in `node_modules/routing-controllers/package.json` and it works! 🎉

**Comment by eduardoweiland** [comment]
That's exactly the problem fixed in PR #48 that is not merged yet.

**Comment by pleerock** [comment]
merged that, hope its fine now
